src/containers/DataDisplay/DataViewTable.tsx

```diff
@@ -1,86 +1,176 @@
 import React, { useEffect, useState } from 'react'
-import { Header, Table, Message } from 'semantic-ui-react'
+import { Header, Table, Message, Search } from 'semantic-ui-react'
 import { Loading } from '../../components'
 import { useLanguageProvider } from '../../contexts/Localisation'
 import usePageTitle from '../../utils/hooks/usePageTitle'
 import { useRouter } from '../../utils/hooks/useRouter'
 import { useDataViewsTable } from '../../utils/hooks/useDataViews'
-import { HeaderRow, DataViewTableAPIQueries } from '../../utils/types'
+import {
+  HeaderRow,
+  DataViewTableAPIQueries,
+  DataViewsTableResponse,
+  BasicStringObject,
+} from '../../utils/types'
 import Markdown from '../../utils/helpers/semanticReactMarkdown'
 import { constructElement, formatCellText } from './helpers'
 import PaginationBar from '../../components/List/Pagination'
+import buildQueryFilters from '../../utils/helpers/list/buildQueryFilters'
+import useDebounce from '../../formElementPlugins/search/src/useDebounce'
+import { useToast } from '../../contexts/Toast'
+import ListFilters from '../List/ListFilters/ListFilters'
+
+interface GqlFilterObject {
+  search?: string
+  [key: string]: any
+}
 
 const DataViewTable: React.FC = () => {
   const { strings } = useLanguageProvider()
   const {
-    push,
     query,
+    location,
+    updateQuery,
     params: { dataViewCode },
   } = useRouter()
 
-  const [apiQueries, setApiQueries] = useState<DataViewTableAPIQueries>({})
-  const { dataViewTable, loading, error } = useDataViewsTable({
+  const [apiQueries, setApiQueries] = useState<BasicStringObject>({})
+  const [gqlFilter, setGqlFilter] = useState<GqlFilterObject>({})
```

Comment:
The API queries contain pagination and sorting (orderBy) info, which gets sent to back-end as url query parameters.

The gqlFilter object contains search data, but will also contain the "filter" restrictions in a single GraphQL filter object.